### PR TITLE
[loader] don't fail with no platforms present

### LINKED ISSUE #518: urInit with no adapters
Currently `urInit` would return UR_ERROR_UNINITIALIZED when no adapters were discovered. I suggest that we either 1) return a more descriptive error e.g. UR_ERROR_NO_ADAPTERS, or 2) return success, and then return 0 platforms in succeeding calls to `urPlatformGet`, if any.

### PR DESCRIPTION
**Pull request by pbalcer**
Closes #518

### CODE REVIEW

File: scripts/templates/ldrddi.cpp.mako [hunk 1]
```diff
@@ -233,35 +224,21 @@ ${tbl['export']['name']}(
...
-    if(!atLeastOneplatformValid)
-        result = ${X}_RESULT_ERROR_UNINITIALIZED;
-    else
-        result = ${X}_RESULT_SUCCESS;
-    %endif
-
     if( ${X}_RESULT_SUCCESS == result )
     {
-        if( ( ur_loader::context->platforms.size() > 1 ) || ur_loader::context->forceIntercept )
+        if( ur_loader::context->platforms.size() == 0 || ur_loader::context->platforms.size() > 1 || ur_loader::context->forceIntercept )
```
Comment (kswiecicki) [inline scripts/templates/ldrddi.cpp.mako:241]: `ur_loader::context->platforms.size() >= 0`

Comment (pbalcer) [reply scripts/templates/ldrddi.cpp.mako:241]: No, this code is here to make direct calls to the adapter if there's only one. I did `... != 1` instead ;)
Thanks. Done.

File: test/loader/platforms/platforms.cpp [hunk 2]
```diff
@@ -0,0 +1,78 @@
+/*
+ *
+ * Copyright (C) 2023 Intel Corporation
+ *
+ * SPDX-License-Identifier: MIT
+ *
+ */
+#include <cassert>
+#include <cstdlib>
```
Comment (PatKamin) [inline test/loader/platforms/platforms.cpp:9]: Include either cstdlib or stdlib.h, certainly there is no need to include both headers.

File: test/loader/platforms/platforms.cpp [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+#include <iostream>
+#include <logger/ur_logger.hpp>
```
Comment (PatKamin) [inline test/loader/platforms/platforms.cpp:11]: nit: Please, make two separate groups of headers: one group with headers from the standard library and the other with UR headers.

**Review by PatKamin — APPROVED** [review]

**Pull request merged**